_posts/2016-05-26-Rust-1.9.md

```diff
@@ -0,0 +1,155 @@
+
+```rust
+let s = "some string".to_string();
+let s = "some string".to_owned();
+```
+
+The former is the official convention, but it was discovered that it's not
+as fast as the second method. For a long time, it was not possible to improve
+this situation, however, a new language feature has landed on nightly and made
+this possible. [`libstd` now uses this internally][str] to make the two equivalent.
+
+[str]: https://github.com/rust-lang/rust/pull/32586
+
+Finally, this feature is technically a library change, but it's worth
+mentioning here as well. The `std::panic` module is being stabilized. This
+allows panics to be caught:
+
+```rust
+use std::panic;
+
+let result = panic::catch_unwind(|| {
+    println!("hello!");
+});
+assert!(result.is_ok());
+
+let result = panic::catch_unwind(|| {
+    panic!("oh no!");
+});
+assert!(result.is_err());
+```
+
+This behavior was defined in [RFC 1236].
+
+[`std::panic`]: http://doc.rust-lang.org/stable/std/panic/index.html
+[RFC 1236]: https://github.com/rust-lang/rfcs/pull/1236
+
+Historically, this has not been possible in Rust: panics are always
+unrecoverable. And in practice, this is still how they should be used. While
+the current implementation of `panic!` does unwinding, in a future release,
+crates will be able to choose a different implementation, namely, that
+`panic!` causes an abort, rather than unwind. Given that panics are for
+non-recoverable errors only, this is a valid implementation.
+
+If panics are for non-recoverable errors, why provide an interface for
+recovering from them? There are two big use-cases here:
+
+* Embedding Rust in other languages
+* Abstractions that manage threads
+
+For the first case, unwinding across a language boundary is undefined behavior,
+and often leads to segfaults in practice. This means that when embedding Rust
+inside of other languages, you must avoid any code that can possibly panic, or
+rely on the (still unstable and only recently landed) 'abort' implementation of
+panic. Allowing panics to be caught at the language boundary provides more
+options in this case: an issue in Rust can return an error, rather than aborting
+the entire process. This is very important for many use-cases of embedding Rust;
+aborting is not always an acceptable option.
+
+In the second, consider a threadpool library. If a thread panics, it's
+reasonable for the pool to not want to abort, but instead report the error,
+and, depending on the details, possibly even restart the thread, in an
+Erlang-style model. Recovering from these kinds of errors is very useful.
+
+This change had a lot of discussion and thought put into it, by a large group
+of people. A significant worry about this change is that people may use it to
+try and emulate exceptions, which is not the intention of this functionality.
+However, the use-cases for this feature are legitimate, and there are two
+reasons we believe that this will not happen, generally:
```

Comment:
This may be getting a little into the weeds here as part of the release announcement, although this seems like information that would ideally live somewhere still. It may be best to just link to the RFC/issue/documentation for now, though, and we could perhaps take an action item to update that with info like this.